collect_app/src/main/java/org/odk/collect/android/widgets/RangeDecimalWidget.java

```diff
@@ -18,20 +18,56 @@
 
 import android.annotation.SuppressLint;
 import android.content.Context;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.TextView;
 
+import androidx.annotation.NonNull;
+
+import com.google.android.material.slider.Slider;
+
+import org.javarosa.core.model.RangeQuestion;
 import org.javarosa.core.model.data.DecimalData;
 import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
+import org.odk.collect.android.views.TrackingTouchSlider;
+import org.odk.collect.android.widgets.utilities.RangeWidgetUtils;
 
 import java.math.BigDecimal;
 
 @SuppressLint("ViewConstructor")
-public class RangeDecimalWidget extends RangeWidget {
+public class RangeDecimalWidget extends QuestionWidget implements Slider.OnChangeListener {
+    private RangeQuestion rangeQuestion;
+    private BigDecimal actualValue;
+
+    TrackingTouchSlider slider;
+    TextView currentValue;
 
     public RangeDecimalWidget(Context context, QuestionDetails prompt) {
         super(context, prompt);
     }
 
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        rangeQuestion = (RangeQuestion) getFormEntryPrompt().getQuestion();
+        RangeWidgetUtils.RangeWidgetLayoutElements layoutElements = RangeWidgetUtils.setUpLayoutElements(context, prompt);
```

Comment:
I think in future this could probably be extracted to its own custom view but for the moment it's great progress!

Reply:
Yes correct @seadowg Thank you!